src/content_encoding.rs

```diff
@@ -0,0 +1,38 @@
+use crate::error::{error, no_error, ErrorIterator, ValidationError};
+use serde_json::Value;
+use std::collections::HashMap;
+
+pub(crate) type ContentTypeCheckType = for<'a> fn(&'a Value, &str) -> ErrorIterator<'a>;
+pub(crate) type ContentTypeConverterType =
+    for<'a> fn(&'a Value, &str) -> Result<String, ValidationError<'a>>;
+
+pub fn is_base64<'a>(instance: &'a Value, instance_string: &str) -> ErrorIterator<'a> {
+    if base64::decode(instance_string).is_err() {
+        return error(ValidationError::format(instance, "base64"));
+    }
+    no_error()
+}
+
+pub fn from_base64<'a>(
+    instance: &'a Value,
+    instance_string: &str,
+) -> Result<String, ValidationError<'a>> {
+    match base64::decode(instance_string) {
+        Ok(value) => Ok(String::from_utf8(value)?),
+        Err(_) => Err(ValidationError::format(instance, "base64")),
+    }
+}
+
+lazy_static::lazy_static! {
+    pub(crate) static ref DEFAULT_CONTENT_ENCODING_CHECK: HashMap<&'static str, ContentTypeCheckType> = {
```

Comment:
Hm, maybe `DEFAULT_CONTENT_ENCODING_CHECKS`? to reflect that there could be many checks